tests/unit/initiateMiddleware.js

```diff
@@ -349,12 +420,265 @@ tap.test('Initiate Middleware', {autoend: true}, t => {
       }
       const endpoint = defaultEndpoint
 
+      // stub for the call to insert data into mongo
+      const stub = sandbox
+        .stub(dbServicesState, 'createEndpointState')
+        .resolves(
+          new Promise(resolve => {
+            resolve({
+              system: {
+                timestamps: {
+                  endpointDuration: {
+                    milliseconds: 102
+                  },
+                  endpointStart: '2020-04-20T13:31:11.301Z',
+                  endpointEnd: '2020-04-20T13:31:11.403Z'
+                }
+              },
+              _id: '5e9da41f963bbe377dee075c',
+              _endpointReference: '5e99657dc4b9120472fdf4eb'
+            })
+          })
+        )
+
       try {
-        updateEndpointState(ctx, endpoint)
+        await updateEndpointState(ctx, endpoint)
         t.pass()
       } catch (err) {
         t.fail()
       }
+
+      // verify stub was called
+      t.ok(stub.called)
+    })
+  })
+
+  t.test('getEndpointByPath', {autoend: true}, t => {
+    t.test('should return the endpoint for the supplied path', async t => {
+      t.plan(3)
+
+      const endpointCache = [
+        {
+          _id: 'endpoint1',
+          name: 'Endpoint 1',
+          endpoint: {
+            pattern: '/path'
+          }
+        },
+        {
+          _id: 'endpoint2',
+          name: 'Endpoint 2',
+          endpoint: {
+            pattern: '/path-2'
+          }
+        }
+      ]
+
+      const endpointCacheMockRevert = initiate.__set__(
+        'endpointCache',
+        endpointCache
+      )
+
+      const endpoint = getEndpointByPath('/path')
+
+      endpointCacheMockRevert()
+
+      t.equal(endpoint._id, 'endpoint1')
+      t.equal(endpoint.name, 'Endpoint 1')
+      t.equal(endpoint.endpoint.pattern, '/path')
+    })
+
+    t.test('should return null if path not found', async t => {
+      t.plan(1)
+
+      const endpointCache = [
+        {
+          _id: 'endpoint1',
+          name: 'Endpoint 1',
+          endpoint: {
+            pattern: '/path'
+          }
+        },
+        {
+          _id: 'endpoint2',
+          name: 'Endpoint 2',
+          endpoint: {
+            pattern: '/path-2'
+          }
+        }
+      ]
+
+      const endpointCacheMockRevert = initiate.__set__(
+        'endpointCache',
+        endpointCache
+      )
+
+      const endpoint = getEndpointByPath('/path-doesnt-exist')
+
+      endpointCacheMockRevert()
+
+      t.notOk(endpoint) // null
+    })
+  })
+
+  t.test('initiateContextMiddleware', {autoend: true}, t => {
+    t.test(
+      'should return with a ctx.status of 404 when endpoint not found',
+      async t => {
+        t.plan(3)
+
+        const ctxMock = {
+          request: {
+            headers: {},
+            path: '/path'
+          },
+          response: {}
+        }
+        const noop = () => {}
```

Comment:
Minor, but maybe the function could stay named `next`, just to make its use in the middleware instantiation more obvious

Reply:
done in 082ccc2

Reply:
and in 787d21b